### ISSUE TITLE
Documentation for methods, hooks etc

### ISSUE THREAD

**Issue by andreydos** [propellant/doctor#2]
I there any possibility to document vuejs hooks, data fields, methods?

**Comment by pearofducks** [comment]
For this I've been using the `description` field; typically, notable elements from the rest of a component go there. I've found not every method needs to be Documented for the rest of the team, it's enough for it to have an inline comment.

I'm open to other suggestions for how to document these, so far most of the ideas I've come up with would deviate from _propdoc's_ core capability (namely being a drop-in component that reads other components, rather than some external service that creates static documents).

**Comment by cbo317110** [comment]
I agree that is not all method to need documentation, but sometimes few of them are very complex. So, why not? You both see any road to archive this feature?

**Comment by pearofducks** [comment]
Outside of having a generic field such as `description` - the only other option would be several keys such as `hooksDocs`, `dataDocs`, `methodsDocs`.

I'm not against that, but won't have time to look into it in the near future.

**Comment by willvincent** [comment]
Why not just use jsdoc in comments on those methods/etc? That is, afterall the purpose of jsdoc, doesn't seem like there's a compelling reason to reinvent that wheel to me.

**Comment by pearofducks** [comment]
That could work as well, and I'm happy to merge a PR adding it - currently this project is focused on parsing/formatting compiled Vue component data, and I don't know that jsdoc is available from that source.

**Comment by cbo317110** [comment]
With a quick search i found that:
https://github.com/Kocal/jsdoc-vuejs
